src/exo/pyparser.py

```diff
@@ -133,9 +133,10 @@ class Parser:
     def __init__(
         self,
         module_ast,
-        func_globals,
-        srclocals,
         getsrcinfo,
+        is_fragment=False,
+        func_globals=None,
+        srclocals=ChainMap(),
```

Comment:
This isn't safe because `ChainMap` is mutable... this should be `None` and then you should have something like `srclocals = srclocals or ChainMap()` in the constructor body.

The issue is that the values of default arguments are stored in the _function_ object and so are shared across all calls. If something gets added to the base chain map in one call, it will be seen by all subsequent calls.

Reply:
Ah ok that makes sense!